### PR TITLE
Drain pipeline queue on pool Put()

### PR DESCRIPTION
**Pull request by adamonduty**
Hi,

Thanks for radix! I've been really happy using it in my project.

I ran into a small issue with the pool. In my project, I use a `limit` parameter to limit the number of items returned. Sometimes I'll pipeline a number of requests to redis (say, 1000) but only call `GetReply()` on the first 100, based on either the `limit` or some other condition. After doing this, I return the redis connection back to the pool. This leads to a weird situation where I see previous replies because the `Append()` and `GetReply()` calls are no longer in sync.

In my project, this was easy to track down, but I imagine it could be a puzzling source of bugs where complex logic was involved. Please review the attached pull and let me know if you'd like any tweaks.

### CODE REVIEW

**Comment by mediocregopher** [comment]
Hey adamonduty! I really appreciate the effort you've taken in making the PR and submitting it, but I don't think I can merge in this particular one. The use-case it corrects for is a strange one, definitely not a common case. I'd prefer to leave the behavior of pool extremely simple and not have it do extra, unexpected things except where absolutely necessary.

For what it's worth the code looks totally fine. For your particular project you might make a little wrapper around pool which implements this particular behavior, something like:

```go
type limitedPool struct {
    *pool.Pool
}

func (lp *limitedPool) Put(conn *redis.Client) {
    // your logic hear
    p.Pool.Put(conn)
}
```

(You might have to re-implement `CarefullyPut`, if you're using that)

Again I'm really sorry to shoot you down like this, and I hope this doesn't dissuade you from contributing in the future :)

**Comment by adamonduty** [comment]
Thanks for the feedback, and especially for responding quickly instead of letting it sit indefinitely! My code is simple enough that I just defer a call to my own drain() method and it works fine.

Would you consider a `DrainPipeline` method on the client instead? As far as I can tell, there is no way to get information of the state of the pipeline, and at least there we'd have access to the length of `Client.pending`.

**Comment by mediocregopher** [comment]
Again, sorry, but probably not. It's another thing I don't think most people need, and the main goal of this package is for simplicity above all else. The state of the pipeline isn't really unknown either, since you know how many commands you've put in and how many you've gotten out (although in your case this might be a bit hard to track, again a wrapper struct may help), so there's no *new* information the client can contribute.

Basically, unless information or functionality which isn't available in some form to the user is needed (i.e. they can't write a wrapper to achieve the same thing), it's really unlikely to be added =/
